### ISSUE TITLE
CPU Load ~100%

### ISSUE THREAD

**Issue by HCMUC** [jmmcatee/cracklord#98]
After finishing a Cracking-job with hashcat or Scanning-job with nmap, top shows a CPU load of 100% for 'cracklord-resourced'.

I wrote a small plugin by myself. When it is finishes the cpu load is normal!

The only errors i found (even in Debug mode) were:

```INFO[0000] Hashcat tool successfully setup```
```INFO[0000] NMap tool successfully setup```
```2015/10/16 11:25:08 method AddTool has wrong number of ins: 2```
```2015/10/16 11:25:08 method Lock has wrong number of ins: 1```
```2015/10/16 11:25:08 method RLock has wrong number of ins: 1```
```2015/10/16 11:25:08 method RLocker has wrong number of ins: 1```
```2015/10/16 11:25:08 method RUnlock has wrong number of ins: 1```
```2015/10/16 11:25:08 method Unlock has wrong number of ins: 1```
```INFO[0000] Listening for queueserver connection.         ip=0.0.0.0 port=9443```

**Comment by jmmcatee** [comment]
This is a known bug. We need to run cracklord with the Go profiler enabled
to identify where it is spending its time.

Michael

**Comment by jmmcatee** [comment]
My guess is this comes from us running loops to copy the buffers of the exec.Cmd Stdout and Stderr. I'm going to play with some code to do this only during a status call and see if that helps.

**Comment by HCMUC** [comment]
If you didn't already know, those two goroutines are causing the high CPU load.

    go func() {
		for {
			io.Copy(v.stderr, v.stderrPipe)
		}
	}()

	go func() {
		for {
			io.Copy(v.stdout, v.stdoutPipe)
		}
	}()

**Comment by jmmcatee** [comment]
I assumed. Did you do a profiling of them to confirm? I have an idea how to
fix it but need to thoroughly test as the last time I tried to do it it
broke everything.

**Comment by HCMUC** [comment]
Sorry i didn't do a profiling of them. I tried pprof but there were some errors. I just removed those two functions to check :/.

I don't know if this is a great fix but it works (tried it with hashcat).
At least the CPU load isn't shown in 'top' and everything like progress or speed is shown.

	go func() {
		for v.job.Progress != 100.00 {
			io.Copy(v.stderr, v.stderrPipe)
		}
	}()
	go func() {
		for v.job.Progress != 100.00 {
			io.Copy(v.stdout, v.stdoutPipe)
		}
	}()

Just a question to nmap: For me the progress diagram is bricked. 
Actually stdout (even in original files) doesn't show "estimated ..." etc
if i run `/usr/bin/cracklord-resourced --conf="/etc/cracklord/resourced.conf"` and hit "ENTER" it prints the stuff like "SYN .. estimated.... etc." and the progress bar is getting updated.

Before it worked but now? I restored the original files but nothing changed... Does it need a flush?
